modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -4529,3 +4529,14 @@ def compare(self, other, **kwargs):
                 other._modin_frame,
             )
         )
+
+    def case_when(self, caselist):
+        qc_type = BaseQueryCompiler
```

Comment:
Why was this changed? Why doesn't `qc_type = type(self)` work?

Reply:
this should be [fixed in tests](https://github.com/modin-project/modin/pull/6972/files#r1531803079) then, not in QC